Author: Andreas Beckmann <debian@abeckmann.de>
Description: Select the correct nv-kernel.o for the target architecture
 The Debian nvidia-kernel-source package supports building both i386 and
 amd64 kernels from one source by including both binary objects. This patch
 makes the build system select the correct one depending on the kernel
 architecture the module is built for.

--- a/Kbuild
+++ b/Kbuild
@@ -43,7 +43,13 @@
 # understands that we want a module.
 #
 
-CORE_OBJS := nv-kernel.o
+ifeq ($(CONFIG_X86_64),y)
+NVARCH = .x86_64
+else
+NVARCH = .i386
+endif
+
+CORE_OBJS := nv-kernel.o$(NVARCH)
 RESMAN_GLUE_OBJS := $(patsubst %.c,%.o,nv.c nv-acpi.c nv-chrdev.c nv-cray.c nv-gvi.c nv-i2c.c nv-mempool.c nv-mlock.c nv-mmap.c nv-p2p.c nv-pat.c nv-procfs.c nv-usermap.c nv-vm.c nv-vtophys.c os-agp.c os-interface.c os-mtrr.c os-registry.c os-smp.c os-usermap.c)
 
 $(MODULE_NAME)-objs := $(CORE_OBJS) $(RESMAN_GLUE_OBJS)
